### DISCUSSION TITLE
"file does not exist"

### DISCUSSION THREAD

**Discussion by Gigamick** [spatie/simple-excel#78] [Q&A] [answered]
First off... thank god for Spatie!

Secondly... getting a really annoying error here (file does not exist). Here is my controller:

```
  $path = $request->file('file')->storeAs('public/csvs', 'contacts.csv');

        // $rows is an instance of Illuminate\Support\LazyCollection
        $rows = SimpleExcelReader::create($path, 'csv')->getRows();

        $rows->each(function (array $rowProperties) {
            dd($rowProperties);
            // in the first pass $rowProperties will contain
            // ['email' => 'john@example.com', 'first_name' => 'john']
        });
```

if I DD out $path I get `public/csvs/contacts.csv` and if I look in my local folders I can see the file is there.. well its actually in public/storage/csvs/contacts.csv but I think thats just how laravel works.

Any ideas?

**Comment by Gigamick** [comment, marked as answer]
For anyone stumbling on this

SimpleExcelReader needs the full path so just prefix with storage_path

```
$rows = SimpleExcelReader::create(storage_path('app/') . $path, 'csv')->getRows();
```

**Reply by Gigamick** [reply]
Credit to @snapey over at Laracasts for this one.

**Reply by freekmurze** [reply]
👍 Feel free to send a PR to the readme to make it more clear.

**Reply by adriel** [reply]
Kinda related (at least this is how I got to this convo). In Laravel, if you're trying to read a file, you can get the full file path with `Storage::path('file.csv');`
https://laravel.com/docs/9.x/filesystem#file-paths
